### PR TITLE
Proxy POST requests to OTP

### PR DESCRIPTION
**Pull request by leonardehrenfried**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

GraphQL queries, which we want to start using soon, almost always require a POST request. For this reason we need to also support them in the middleware which this PR achieves.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java [hunk 1]
```diff
@@ -83,17 +97,85 @@ public void bind(final SparkSwagger restApi) {
                 .withDescription("Forwards any GET request to " + otpVersion.toString() + ". " + OTP_DOC_LINK)
                 .withQueryParam(USER_ID)
                 .withProduces(List.of(MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML)),
-                this::proxy
+                this::proxyGET
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java]: I prefer `proxyGet`.

File: src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java [hunk 2]
```diff
@@ -83,17 +97,85 @@ public void bind(final SparkSwagger restApi) {
...
+        ).post(path("/*")
+                .withDescription("Forwards any POST request to " + otpVersion.toString() + ". " + OTP_DOC_LINK)
+                .withQueryParam(USER_ID),
+                this::proxyPOST
```
Comment (binh-dam-ibigroup) [inline src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java]: I prefer `proxyPost`.
Also, can you add the expected response type? (see `.withProduces` in the GET case)

**Review by binh-dam-ibigroup — APPROVED** [review]
Approving subject to adding the output MIME type for the POST methods.

File: src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java (hunk 2 above)
Comment (leonardehrenfried) [reply src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java]: Done.

File: src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java (hunk 1 above)
Comment (leonardehrenfried) [reply src/main/java/org/opentripplanner/middleware/controllers/api/OtpRequestProcessor.java]: Renamed both methods.

**Review by br648 — APPROVED** [review]
LGTM!

**Pull request merged**